### PR TITLE
Getting codecov to work on every PRs

### PR DESCRIPTION
**Pull request by sebastienvas**
1. Pushing code coverage metrics regardless of failing tests. 
2. Re-enabling codecov in pre and post submit.
3. Adding a file to add pkg to skip
4. Merging parallel-codecov and codecov
5. Merging all code coverage as discussed in test-release meeting

### CODE REVIEW

**Comment by sebastienvas** [comment]
@GregHanson we decided to have global code coverage in the test-release meeting. So it is undoing a little bit what you did. The good news is that now code coverage is actually working and we'll be able to gate it.

File: .codecov.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 coverage:
+  precision: 0
+  round: up
```
Comment (ldemailly) [inline .codecov.yml]: *cough* *cough*

File: codecov.skip [hunk 2]
```diff
@@ -0,0 +1,3 @@
+istio.io/istio/tests*
+istio.io/istio/mixer/tools/codegen*
+istio.io/istio/security/tests/integration*
```
Comment (ldemailly) [inline codecov.skip]: why not those?

**Comment by ldemailly** [comment]
on circle:
The following tests failed
FAIL: istio.io/istio/mixer/test/client/stress

https://circleci.com/gh/istio/istio/36844?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

/lgtm

**Comment by ldemailly** [comment]
/hold
in case you want to fix the above
otherwise feel free to merge

**Comment by sebastienvas** [comment]
I think this test is flaky. I'll file a bug for this.
/unhold

File: codecov.skip (hunk 2 above)
Comment (sebastienvas) [reply codecov.skip]: the first 2 were already skipped, I added the integration one because they need a real cluster.

**Pull request merged**